protocol-units/settlement/mcr/contracts/script/helpers/Helper.sol

```diff
@@ -0,0 +1,357 @@
+                    address(safeSingleton),
+                    address(fallbackHandler),
+                    config.signersLabs,
+                    config.thresholdLabs
+                );
+                deployment.movementFoundationSafe = _deploySafe(
+                    safeFactory,
+                    address(safeSingleton),
+                    address(fallbackHandler),
+                    config.signersFoundation,
+                    config.thresholdFoundation
+                );
+            }
+        }
+        console.log("Safe addresses:");
+        console.log("Deployer:", address(deployment.movementDeployerSafe));
+        console.log("Labs:", address(deployment.movementLabsSafe));
+        console.log("Foundation:", address(deployment.movementFoundationSafe));
+    }
+
+    function _deploySafe(
+        SafeProxyFactory safeFactory,
+        address safeSingleton,
+        address fallbackHandler,
+        address[] memory signers,
+        uint256 threshold
+    ) internal returns (address safe) {
+        safe = payable(
+            address(
+                safeFactory.createProxyWithNonce(
+                    safeSingleton,
+                    abi.encodeWithSignature(
+                        safeSetupSignature, signers, threshold, ZERO, "0x", fallbackHandler, ZERO, 0, payable(ZERO)
+                    ),
+                    0
+                )
+            )
+        );
+    }
+
+    function _deployTimelock() internal {
+        if (deployment.timelock == ZERO) {
+            timelock = new TimelockController(config.minDelay, config.signersLabs, config.signersFoundation, ZERO);
+            deployment.timelock = address(timelock);
+        }
+    }
+
+    function _deployCreate3() internal {
+        if (address(create3).code.length == 0) {
+            console.log("CREATE3: deploying");
+            create3 = new CREATE3Factory();
+        }
+    }
+
+    function _deployDependencies() internal {
+        _deployCreate3();
+        _deploySafes();
+        _deployTimelock();
+    }
+
+    function _storeAdminDeployment() internal returns (address admin) {
+        Vm.Log[] memory logs = vm.getRecordedLogs();
+        admin = logs[logs.length - 2].emitter;
+        console.log("admin", admin);
+    }
+
+    function _writeDeployments() internal {
+        string memory path = string.concat(root, deploymentsPath);
+        string memory json = storageJson;
+        string memory base = "new";
+        string memory newChainData = _serializer(json, deployment);
+        // take values from storageJson that were not updated (e.g. 3771) and serialize them
+        uint256[] memory validChains = new uint256[](4);
+        validChains[0] = 1; // ethereum
+        validChains[1] = 11155111; // sepolia
+        validChains[2] = 17000; // holesky
+        validChains[3] = 31337; // foundry
+        for (uint256 i = 0; i < validChains.length; i++) {
+            if (validChains[i] != block.chainid) {
```

Comment:
Explain the purpose of this. This is going to run over all the valid chains and serialize for every one that is not the current chain? Why?